src/bin/pg_dump/dump_babel_utils.c

```diff
@@ -1574,3 +1574,135 @@ fixCursorForBbfTableData(Archive *fout,
 	*nfields_new = fixCursorForBbfSqlvariantTableData(fout, tbinfo, buf, *nfields,
 													  sqlvar_metadata_pos);
 }
+
+/*
+ * babelfishDumpOpclassHelper - This particular helper function would be helpful
+ * to dump Babelfish operator class created on the top of built-in data types of
+ * the Postgres. For example, Operator class sys.int_numeric for the type int4.
+ * This is needed because of some bug in postgres due to which Operarors and
+ * Support functions are not being dumped correctly for user defined operator
+ * class defined for built-in datatypes.
+ */
+void
+babelfishDumpOpclassHelper(Archive *fout, const OpclassInfo *opcinfo, PQExpBuffer *buff, bool *needComma)
+{
+	PQExpBuffer	query;
+	PGresult	*res;
+	int			i_opfnamespace;
+	int			i_opfname;
+	char		*opfnamespace;
+	char		*opfname;
+	const char	*str;
+	const char 	*opclass = fmtQualifiedDumpable(opcinfo);
+
+
+	/* Firstly check that Operator class is part of Babelfish */
+	if (strcasecmp(opclass, "\"sys\".\"int_numeric\"") != 0 &&
+		strcasecmp(opclass, "\"sys\".\"numeric_int\"") != 0 &&
+		strcasecmp(opclass, "\"sys\".\"int2_numeric\"") != 0 &&
+		strcasecmp(opclass, "\"sys\".\"numeric_int2\"") != 0 &&
+		strcasecmp(opclass, "\"sys\".\"int8_numeric\"") != 0 &&
+		strcasecmp(opclass, "\"sys\".\"numeric_int8\"") != 0)
```

Comment:
The identifiers might not be quoted depending upon `quote_all_identifiers` setting. Let's incorporate both the cases of identifiers being quoted and not quoted.

Reply:
Addressed in new revision